enrichment-web/src/main/java/eu/europeana/enrichment/web/service/impl/EnrichmentTopicServiceImpl.java

```diff
@@ -59,18 +86,21 @@ public Topic createTopic(Topic topic) throws HttpException, UnsupportedEntityTyp
 		return topic;
 	}
 
+	
 	@Override
-	public Topic updateTopic(Topic topic) {
-		Topic dbtopicEntity = persistentTopicService.getByIdentifier(topic.getIdentifier());
+	public Topic updateTopic(Long topicId, Topic topic) {
+		Topic dbtopicEntity = persistentTopicService.getByTopicId(topicId);
 		if (dbtopicEntity != null)
 		{
+			dbtopicEntity.setBaseURI(configuration.getEnrichApiEndpoint());
+			dbtopicEntity.getModel().setBaseURI(configuration.getEnrichApiEndpoint());
 			if (topic.getTerms() != null)
 				dbtopicEntity.setTerms(topic.getTerms());
 			if (topic.getKeywords() != null)
 				dbtopicEntity.setKeywords(topic.getKeywords());
 			if (topic.getDescriptions() != null)
 				dbtopicEntity.setDescriptions(topic.getDescriptions());
-			if (topic.getTopicID() != null)
+			if (topic.getTopicID() != 0L)
 				dbtopicEntity.setTopicID(topic.getTopicID());
```

Comment:
Topic id is not to be set here, just used to get the topic, that field must not be changed in this request.